src/adb.js

```diff
@@ -1 +1,353 @@
-// https://github.com/appcelerator/titanium_mobile/blob/master/node_modules/titanium-sdk/lib/adb.js
+import 'babel-polyfill';
+import 'source-map-support/register';
+import fs from 'fs';
+import path from 'path';
+import wrench from 'wrench';
+import { spawn } from 'child_process';
+
+import { Connection } from './connection';
+import { Emulator, EmulatorManager } from './emulator';
+import * as util from './util';
+import * as sdk from './sdk';
+
+/**
+ * Debug flag that is enabled via the android.debugadb setting.
+ */
+let DEBUG = false;
+
+/**
+ * Creates an ADB object.
+ * @class
+ * @classdesc Provides methods to interact with the Android Debug Bridge (ADB).
+ * @constructor
+ */
+export class ADB {
+	constructor(config) {
+		this.config = config;
+		if (config && config.get('android.debugadb', false)) {
+			DEBUG = true;
+		}
+	}
+
+	/**
+	 * Returns the version of the ADB server.
+	 * @returns {Promise}
+	 */
+	version() {
+		return new Connection(this)
+			.exec('host:version')
+			.then(data => {
+				return '1.0.' + parseInt(data, 16);
+			})
+			.catch(err => {
+				console.log('adb version: ', err);
+			});
+	}
+
+	/**
+	 * Runs the specified command on the Android emulator/device. Note that ADB
+	 * converts all \n to \r\n. So data will probably be larger than the original
+	 * output on the device.
+	 * @param {String} cmd - The command to run
+	 * @returns {Promise}
+	 */
+	shell(deviceId, cmd) {
+		const conn = new Connection(this);
+		return conn
+			.exec('host:transport:' + deviceId)
+			.then(data => {
+				return conn.exec('shell:' + cmd.replace(/^shell\:/, ''), { bufferUntilClose: true, noLength: true });
+			})
+			.then(result => result)
+			.catch(err => {
+				console.log('adb shell: ', err);
+			});
+	}
+
+	/**
+	 * Attempts to find the adb executable, then start the adb server.
+	 * @returns {Promise}
+	 */
+	startServer() {
+		return sdk
+			.detect(this.config)
+			.then(result => {
+				return util.run(result.sdk.executables.adb, ['start-server']);
+			}).catch(err => {
+				console.log('start server', err);
+			});
+	}
+
+	/**
+	 * Attempts to find the adb executable, then stop the adb server.
+	 * @returns {Promise}
+	 */
+	stopServer() {
+		return sdk
+			.detect(this.config)
+			.then(result => {
+				return util.run(result.sdk.executables.adb, ['kill-server']);
+			})
+			.catch(err => {
+				console.log('stop server', err);
```

Comment:
We don't want to catch errors and we don't want to `console.log()`. This should be done by the consumer of this library.